test.js

```diff
@@ -29,7 +35,7 @@ test('returns `undefined` when nothing could be found', async t => {
 	t.is((await m([1, 2, 3], () => false)), undefined);
 });
 
-test('rejected return value in `tester` rejects the promise', async t => {
-	const fixtureErr = new Error('fixture');
-	await t.throws(m([1, 2, 3], () => Promise.reject(fixtureErr)), fixtureErr.message);
-});
+// test('rejected return value in `tester` rejects the promise', async t => {
+// 	const fixtureErr = new Error('fixture');
+// 	await t.throws(m([1, 2, 3], () => Promise.reject(fixtureErr)), fixtureErr.message);
+// });
```

Comment:
After looking thoroughly at the code, I believe it's because of the way the `reduce` call chains promises, the ensuing `.catch()` only affects the first rejecting promise of the reducer. Since in the failing test you map 3 elements to a rejecting promise each, you end up with a chain that looks like

```
Rejection -> Catch -> Rejection -> Rejection
```

which explains the two unhandled rejections in that test. If you only return a rejection for one of the 3 elements, the test succeeds. Very subtle. The reducer would probably have to stop chaining promises for the remaining elements as soon as one rejects, or perhaps there is yet an even better solution that doesn't occur to me right now.